user_api.go

```diff
@@ -0,0 +1,447 @@
+}
+
+// Transactions retrieves a paginated list of transactions via the user transactions API.
+// The returned response includes transactions and pagination details, such as the page number,
+// sort order, and sorting field (sortBy).
+//
+// This method allows optional query parameters to be applied via the provided query options.
+// The response is expected to be to unmarshal into a *response.PageModel[response.Transaction] struct.
+// Returns an error if the request fails or the response cannot be decoded.
+func (u *UserAPI) Transactions(ctx context.Context, opts ...queries.TransactionsQueryOption) (*queries.TransactionPage, error) {
+	res, err := u.transactionsAPI.Transactions(ctx, opts...)
+	if err != nil {
+		return nil, transactions.HTTPErrorFormatter("retrieve transactions page", err).FormatGetErr()
+	}
+
+	return res, nil
+}
+
+// Transaction retrieves a specific transaction by its ID via the user transactions API.
+// The response is expected to be unmarshaled into a *response.Transaction struct.
+// Returns an error if the request fails or the response cannot be decoded.
+func (u *UserAPI) Transaction(ctx context.Context, ID string) (*response.Transaction, error) {
+	res, err := u.transactionsAPI.Transaction(ctx, ID)
+	if err != nil {
+		msg := fmt.Sprintf("record a transaction with ID: %s", ID)
+		return nil, transactions.HTTPErrorFormatter(msg, err).FormatGetErr()
+	}
+
+	return res, nil
+}
+
+// XPub retrieves the full xpub information for the current user via the users API.
+// The response is unmarshaled into a *response.Xpub.
+// Returns an error if the request fails or the response cannot be decoded.
+func (u *UserAPI) XPub(ctx context.Context) (*response.Xpub, error) {
+	res, err := u.xpubAPI.XPub(ctx)
+	if err != nil {
+		return nil, users.XPubsHTTPErrorFormatter("retrieve xpub information", err).FormatGetErr()
+	}
+
+	return res, nil
+}
+
+// UpdateXPubMetadata updates the metadata associated with the current user's xpub via the users API.
+// The response is unmarshaled into a *response.Xpub.
+// Returns an error if the request fails or the response cannot be decoded.
+func (u *UserAPI) UpdateXPubMetadata(ctx context.Context, cmd *commands.UpdateXPubMetadata) (*response.Xpub, error) {
+	res, err := u.xpubAPI.UpdateXPubMetadata(ctx, cmd)
+	if err != nil {
+		return nil, users.XPubsHTTPErrorFormatter("update xpub metadata ", err).FormatGetErr()
+	}
+
+	return res, nil
+}
+
+// GenerateAccessKey creates a new access key associated with the current user's xpub via the users access key API.
+// The response is unmarshaled into a *response.AccessKey.
+// Returns an error if the request fails or the response cannot be decoded.
+func (u *UserAPI) GenerateAccessKey(ctx context.Context, cmd *commands.GenerateAccessKey) (*response.AccessKey, error) {
+	res, err := u.accessKeyAPI.GenerateAccessKey(ctx, cmd)
+	if err != nil {
+		return nil, users.AccessKeysHTTPErrorFormatter("generate access key ", err).FormatPostErr()
```

Comment:
I'd stick with one naming of the formatters. I see in some API methods you are using `HTTPErrorFormatter` and here and below you are adding `AccessKeys` while above `XPubs` to `HTTPErrorFormatter`. Any reason for different names here?

Reply:
They are in the same package and therefore can't be named like other formatters. 😞.